hdfs tests failing on main

```python-traceback
______________________ ERROR at setup of test_read_bytes _______________________

request = <SubRequest 'hdfs' for <Function test_read_bytes>>

    @pytest.fixture
    def hdfs(request):
>       hdfs = pyarrow.hdfs.connect(host="localhost", port=8020)

dask/bytes/tests/test_hdfs.py:34: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = 'localhost', port = 8020, user = None, kerb_ticket = None
extra_conf = None

    def connect(host="default", port=0, user=None, kerb_ticket=None,
                extra_conf=None):
        """
        Connect to an HDFS cluster. All parameters are optional and should
        only be set if the defaults need to be overridden.
    
        Authentication should be automatic if the HDFS cluster uses Kerberos.
        However, if a username is specified, then the ticket cache will likely
        be required.
    
        Parameters
        ----------
        host : NameNode. Set to "default" for fs.defaultFS from core-site.xml.
        port : NameNode's port. Set to 0 for default or logical (HA) nodes.
        user : Username when connecting to HDFS; None implies login user.
        kerb_ticket : Path to Kerberos ticket cache.
        extra_conf : dict, default None
          extra Key/Value pairs for config; Will override any
          hdfs-site.xml properties
    
        Notes
        -----
        The first time you call this method, it will take longer than usual due
        to JNI spin-up time.
    
        Returns
        -------
        filesystem : HadoopFileSystem
        """
        warnings.warn(
            _DEPR_MSG.format("hdfs.connect", "2.0.0", "fs.HadoopFileSystem"),
>           DeprecationWarning, stacklevel=2
        )
E       DeprecationWarning: pyarrow.hdfs.connect is deprecated as of 2.0.0, please use pyarrow.fs.HadoopFileSystem instead.

/opt/conda/lib/python3.7/site-packages/pyarrow/hdfs.py:215: DeprecationWarning
```
https://github.com/dask/dask/runs/2731208592

--- a/continuous_integration/hdfs/install.sh
+++ b/continuous_integration/hdfs/install.sh
@@ -1,7 +1,7 @@
 #!/bin/bash
 set -xe
 
-docker exec hdfs conda install -y -q dask pyarrow">=0.14.0,!=2.0.0" fsspec pyyaml -c conda-forge
+docker exec hdfs conda install -y -q dask pyarrow">=0.14.0,!=2.0.0" fsspec pyyaml openjdk -c conda-forge
 docker exec hdfs python -m pip install -e .
 
 set +xe

--- a/dask/bytes/tests/test_hdfs.py
+++ b/dask/bytes/tests/test_hdfs.py
@@ -16,23 +16,21 @@ try:
 except (ImportError, SyntaxError):
     distributed = None
 
-try:
-    import pyarrow
-    from pyarrow.fs import HadoopFileSystem
-except ImportError:
-    pyarrow = None
-
 
 if not os.environ.get("DASK_RUN_HDFS_TESTS", ""):
     pytestmark = pytest.mark.skip(reason="HDFS tests not configured to run")
 
+import pyarrow
+
+# from pyarrow.fs import HadoopFileSystem
+from pyarrow.hdfs import _connect
 
 basedir = "/tmp/test-dask"
 
 
 @pytest.fixture
 def hdfs(request):
-    hdfs = HadoopFileSystem(host="localhost", port=8020)
+    hdfs = _connect(host="localhost", port=8020)
 
     if hdfs.exists(basedir):
         hdfs.rm(basedir, recursive=True)

--- a/dask/bytes/tests/test_hdfs.py
+++ b/dask/bytes/tests/test_hdfs.py
@@ -26,7 +26,8 @@ try:
     from pyarrow.hdfs import _connect, _maybe_set_hadoop_classpath
 except ImportError:
     try:
-        from pyarrow._hdfs import connect as _connect, _maybe_set_hadoop_classpath
+        from pyarrow._hdfs import _maybe_set_hadoop_classpath
+        from pyarrow._hdfs import connect as _connect
     except ImportError:
         pyarrow = False
 

--- a/dask/bytes/tests/test_hdfs.py
+++ b/dask/bytes/tests/test_hdfs.py
@@ -20,7 +20,7 @@ except (ImportError, SyntaxError):
 if not os.environ.get("DASK_RUN_HDFS_TESTS", ""):
     pytestmark = pytest.mark.skip(reason="HDFS tests not configured to run")
 
-import pyarrow
+pyarrow = pytest.importorskip("pyarrow")
 
 # from pyarrow.fs import HadoopFileSystem
 from pyarrow.hdfs import _connect

--- a/dask/bytes/tests/test_hdfs.py
+++ b/dask/bytes/tests/test_hdfs.py
@@ -25,7 +25,7 @@ pyarrow = pytest.importorskip("pyarrow")
 try:
     from pyarrow.hdfs import _connect, _maybe_set_hadoop_classpath
 except ImportError:
-    try
+    try:
         from pyarrow._hdfs import connect as _connect, _maybe_set_hadoop_classpath
     except ImportError:
         pyarrow = False

--- a/dask/bytes/tests/test_hdfs.py
+++ b/dask/bytes/tests/test_hdfs.py
@@ -18,7 +18,7 @@ except (ImportError, SyntaxError):
 
 try:
     import pyarrow
-    from pyarrow.hdfs import HadoopFileSystem
+    from pyarrow.fs import HadoopFileSystem
 except ImportError:
     pyarrow = None
 

--- a/dask/bytes/tests/test_hdfs.py
+++ b/dask/bytes/tests/test_hdfs.py
@@ -22,14 +22,20 @@ if not os.environ.get("DASK_RUN_HDFS_TESTS", ""):
 
 pyarrow = pytest.importorskip("pyarrow")
 
-# from pyarrow.fs import HadoopFileSystem
-from pyarrow.hdfs import _connect
+try:
+    from pyarrow.hdfs import _connect, _maybe_set_hadoop_classpath
+except ImportError:
+    try
+        from pyarrow._hdfs import connect as _connect, _maybe_set_hadoop_classpath
+    except ImportError:
+        pyarrow = False
 
 basedir = "/tmp/test-dask"
 
 
 @pytest.fixture
 def hdfs(request):
+    _maybe_set_hadoop_classpath()
     hdfs = _connect(host="localhost", port=8020)
 
     if hdfs.exists(basedir):

--- a/dask/bytes/tests/test_hdfs.py
+++ b/dask/bytes/tests/test_hdfs.py
@@ -31,7 +31,7 @@ basedir = "/tmp/test-dask"
 
 @pytest.fixture
 def hdfs(request):
-    hdfs = pyarrow.hdfs.connect(host="localhost", port=8020)
+    hdfs = pyarrow.fs.HadoopFileSystem(host="localhost", port=8020)
 
     if hdfs.exists(basedir):
         hdfs.rm(basedir, recursive=True)

--- a/dask/bytes/tests/test_hdfs.py
+++ b/dask/bytes/tests/test_hdfs.py
@@ -18,6 +18,7 @@ except (ImportError, SyntaxError):
 
 try:
     import pyarrow
+    from pyarrow.hdfs import HadoopFileSystem
 except ImportError:
     pyarrow = None
 
@@ -31,7 +32,7 @@ basedir = "/tmp/test-dask"
 
 @pytest.fixture
 def hdfs(request):
-    hdfs = pyarrow.fs.HadoopFileSystem(host="localhost", port=8020)
+    hdfs = HadoopFileSystem(host="localhost", port=8020)
 
     if hdfs.exists(basedir):
         hdfs.rm(basedir, recursive=True)

